portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +28,24 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Adds a random fact to the page.
+ */
+function addRandomFact() {
+    const facts = ["I have no sibling", 
+                   "I seldom eat spicy food", 
+                   "I don't like watching thrillers"];
+    
+    // Pick a random fact.
+    let randomIndex = Math.floor(Math.random() * facts.length);
+    while (randomIndex == old) {
```

Comment:
Practically won't, but theoretically may loop very long if we keep getting the same number.
Since length > 1, maybe we can get next different random by 
next = (current + random (1, length - 1)) modular length?

Maybe can split this method into 2 independent tasks.
getRandomFact() returns a random fact
showFact(fact) shows a given fact?
This way it's easier to write the test for each of the smaller method.

Reply:
Fixed